libs/designsystem/src/lib/components/calendar/calendar.component.spec.ts

```diff
@@ -266,6 +267,138 @@ describe('CalendarComponent', () => {
+      });
+
+      describe('when `minDate` changes to a later date than `todayDate`', () => {
+        let newMinDate: Date;
+
+        beforeEach(() => {
+          newMinDate = new Date(
+            todayDate.getFullYear() + 1,
+            todayDate.getMonth(),
+            todayDate.getDay()
+          );
+          spectator.setInput('minDate', newMinDate);
+        });
+
+        it('should set active month based on changed `minDate`', () => {
+          expect(spectator.component.activeMonthName).toEqual(moment(newMinDate).format('MMMM'));
+        });
+      });
+    });
+
+    describe('when `maxDate` is set', () => {
+      let maxDate: Date;
+      const yearsAfterTodayDate = 2;
+
+      beforeEach(() => {
+        maxDate = new Date(todayDate.getFullYear() + yearsAfterTodayDate, 0, 1);
+        spectator.setInput('maxDate', maxDate);
+      });
+
+      describe('when `maxDate` changes to an earlier date then `todayDate`', () => {
+        let newMaxDate: Date;
+
+        beforeEach(() => {
+          newMaxDate = new Date(
+            todayDate.getFullYear() - 1,
+            todayDate.getMonth(),
+            todayDate.getDay()
+          );
+          spectator.setInput('maxDate', newMaxDate);
+        });
+
+        it('should set active month based on changed `maxDate`', () => {
+          expect(spectator.component.activeMonthName).toEqual(moment(newMaxDate).format('MMMM'));
+        });
+      });
+    });
+  });
+
+  describe('year navigator', () => {
+    describe('by default', () => {
+      it('should not render', () => {
+        expect(spectator.element.querySelector('kirby-dropdown')).toBeNull();
+      });
+    });
+
+    describe('when yearNavigatorOptions are set', () => {
+      let todayDate: Date;
+      let yearNavigatorOptions: CalendarYearNavigatorConfig;
+
+      beforeEach(() => {
+        todayDate = new Date(2021, 0, 1);
+        yearNavigatorOptions = { from: -3, to: 2 };
+
+        spectator.setInput('todayDate', todayDate);
+        spectator.setInput('selectedDate', todayDate);
+        spectator.setInput('yearNavigatorOptions', yearNavigatorOptions);
+      });
+
+      it('should render', () => {
+        expect(spectator.element.querySelector('kirby-dropdown')).not.toBeNull();
+      });
+
+      it('should change year on navigation', () => {
+        const monthsUntilNextDecember = 12 - (todayDate.getMonth() + 1);
+        const firstNavigatedYearIndex = spectator.component.navigatedYear;
+
+        spectator.component.changeMonth(monthsUntilNextDecember);
+        expect(spectator.component.activeYear).toEqual(todayDate.getFullYear().toString());
+        expect(spectator.component.navigatedYear).toEqual(firstNavigatedYearIndex);
+
+        spectator.component.changeMonth(1);
+        expect(spectator.component.activeYear).toEqual((todayDate.getFullYear() + 1).toString());
+        expect(spectator.component.navigatedYear).toEqual(firstNavigatedYearIndex + 1);
+      });
```

Comment:
Consider readability by just setting selectedDate to 31/12/2021 and then navigating 1 month and assert selected year in dropdown is 2022...

Reply:
Implemented. Check out 87e844cc.